plugins/modules/swim_workflow_manager.py

```diff
@@ -937,62 +937,60 @@ def get_device_uuids(self, site_name, device_family, device_role, device_series_
+                        function='get_site_assigned_network_devices',
+                        op_modifies=True,
+                        params={"site_id": site_id},
+                    )
+                    self.log("Received API response from 'get_site_assigned_network_devices': {0}".format(str(response)), "DEBUG")
+                    response = response.get('response')
 
-                if not response:
-                    self.log("No devices found for site '{0}'.". format(site_name), "WARNING")
+                    if not response:
+                        self.log("No devices found for site - '{0}'.". format(site_name), "WARNING")
 
-            except Exception as e:
-                self.log("Unable to fetch the device(s) associated to the site '{0}' due to '{1}'".format(site_name, str(e)), "WARNING")
-                return device_uuid_list
+                    for device_id in response:
+                        device_id_list.append(device_id.get("deviceId"))
 
-            for device_id in response:
-                device_id_list.append(device_id.get("deviceId"))
+                except Exception as e:
+                    self.log("Unable to fetch the device(s) associated to the site '{0}' due to '{1}'".format(site_name, str(e)), "WARNING")
+                    return device_uuid_list
 
-            try:
-                device_params = {}
-                offset = 0
-                limit = self.get_device_details_limit()
-                initial_exec = False
+            for device_id in device_id_list:
+                self.log("Processing device_id: {0}".format(device_id))
+                try:
+                    device_list_response = self.dnac._exec(
+                        family="devices",
+                        function="get_device_list",
+                        params={"id": device_id},
+                    )
 
-                while True:
-                    if initial_exec:
-                        device_params["limit"] = limit
-                        device_params["offset"] = offset * limit
-                        device_list_response = self.dnac._exec(
-                            family="devices",
-                            function='get_device_list',
-                            params=device_params
-                        )
-                    else:
-                        initial_exec = True
-                        device_list_response = self.dnac._exec(
-                            family="devices",
-                            function='get_device_list',
-                            op_modifies=True
-                        )
-                    offset = offset + 1
-                    self.log("Received API response from 'device_list_response': {0}".format(str(device_list_response)), "DEBUG")
-                    device_response = device_list_response.get('response')
+                    self.log("Received API response: {0}".format(str(device_list_response)), "DEBUG")
```

Comment:
self.log("Received API response from 'get_device_list': {0}".format(str(device_list_response)), "DEBUG")

Reply:
fixed the code